app/controllers/api/physical_servers_controller.rb

```diff
@@ -52,9 +52,20 @@ def refresh_resource(type, id, _data = nil)
       end
     end
 
+    # Process apply config pattern operation for single and multi-resources
+    #
+    # Even the request for multi-resources isn't completed successfully, return a HTTP Status 200
+    def apply_config_pattern_resource(type, id, data)
+      ensure_resource_exists(:customization_scripts, data["pattern_id"])
```

Comment:
do we need this method ?  can't we just call resource_search instead ?

Reply:
Yeah, We can just call resource_search, but I think it would make more sense if we used the result of the method to do something. Since we just need to do the check I guess that ensure_resource_script makes sense. What do you think?

Reply:
I can remove it in change_resource_state, but I'll have to raise an exception when the operation is for single resource, like I did in apply_config_pattern_resource in line 62. But I can't remove the ensure_resource_exists method in line 59 because I have to ensure that the config_pattern resource is valid also, and the resource_search just check the physical_server resource.